source/server/listener_manager_impl.cc

```diff
@@ -561,23 +562,58 @@ void ListenerManagerImpl::startWorkers(GuardDog& guard_dog) {
   }
 }
 
+void ListenerManagerImpl::stopListener(Network::ListenerConfig& listener,
+                                       std::function<void()> callback) {
+  const uint64_t listener_tag = listener.listenerTag();
+  const auto workers_pending_stop = std::make_shared<std::atomic<uint32_t>>(workers_.size());
+  for (const auto& worker : workers_) {
```

Comment:
Can you add a few more comments throughout this function? It's a bit dense/nested to quickly understand what is going on.